authority/ssh.go

```diff
@@ -205,59 +220,78 @@ func (a *Authority) GetSSHBastion(ctx context.Context, user string, hostname str
 
 // SignSSH creates a signed SSH certificate with the given public key and options.
 func (a *Authority) SignSSH(ctx context.Context, key ssh.PublicKey, opts provisioner.SignSSHOptions, signOpts ...provisioner.SignOption) (*ssh.Certificate, error) {
-	var mods []provisioner.SSHCertModifier
-	var validators []provisioner.SSHCertValidator
+	var (
+		certOptions []sshutil.Option
+		mods        []provisioner.SSHCertModifier
+		validators  []provisioner.SSHCertValidator
+	)
+
+	// Validate given options.
+	if err := opts.Validate(); err != nil {
+		return nil, errs.Wrap(http.StatusBadRequest, err, "authority.SignSSH")
+	}
 
 	// Set backdate with the configured value
 	opts.Backdate = a.config.AuthorityConfig.Backdate.Duration
 
 	for _, op := range signOpts {
 		switch o := op.(type) {
+		// add options to NewCertificate
+		case provisioner.SSHCertificateOptions:
```

Comment:
At some point I went through and changed as many instances of `Certificate` -> `Cert` as I could find. We should choose one or the other.

Reply:
If `Cert` always means certificate in our code base, then there's probably no need to use the longer version (even if conforms with golang x509/ssh). Just my opinion. Either way, we should be consistent.

Reply:
I'm not changing it right now.